Added format strings for well known formats

MarkDDR opened the pull request:
Filled out the "Well Known Format Descriptions" chapter with datetime samples as well as format string samples if someone wanted an example of what they would look like for those formats.

jhpratt commented:
A few nits:

- It should be called a _format description_, not a _format string_.
- It should be noted that the format descriptions are only approximations. This is because there is a stupid amount of subtleties, obsolete syntax, etc. mandated by the standards. 
- The format description for ISO 8601 is only the default configuration. Unlike the others, it is configurable to an extent.

Just because, here's an example of something that is technically valid RFC2822. Please don't include this in the documentation, though — I just wanted to share the absurdities.

```rust
#[rustfmt::skip]
assert_eq!(
    Time::parse(
        "  \t Sat,\r\n \
        (\tfoo012FOO!)\
        (\u{1}\u{b}\u{e}\u{7f})\
        (\\\u{1}\\\u{9}\\\u{28}\\\u{29}\\\\u{5c}\\\u{7f})\
        (\\\n\\\u{b})\
        02 \r\n  \r\n Jan 2021 03:04:05 GMT",
        &Rfc2822
    )?,
    time!(03:04:05)
);
```

Other than these minor things, LGTM! Thank you!

MarkDDR commented:
Thanks for the feedback, I adjusted the phrasing, added a paragraph up top explaining that they are approximations for learning purposes, and also explained that ISO 8601 can be configured. It would probably make sense for the book to more thoroughly explain what ISO 8601's various configuration options look like but I'll leave that for a future pull request. For now I'm planning on turning my attention toward adding more code examples and explanations to the API docs themselves as that is currently looking a tad barren in places where a couple examples would really help

jhpratt commented:
Looks good. Thanks!
